### DISCUSSION TITLE
CSV Products Import

### DISCUSSION THREAD

**Discussion by mraf** [givanz/Vvveb#164] [General]
Hey,

is it better to create plugin for importing csv file with products or we can follow and use the same idea for import as Xml or Sql in import controller?

**Comment by givanz** [comment]
Hi

For CSV import a mapping for fields like title, image etc is needed with some automatic detection and an interface for the user to correct the mapping if needed.

The plugin can either generate xml files than can be used with the xml importer or generate the insert sqls directly.

The xml importer can handle multi table data by keeping references for table id's, the sql importer will run the queries directly without any processing.

With csv things can get complex for secondary tables like product options, multi language product description, reviews etc to keep track of id's.
A simple way is to use a blank installation and insert all the id's from the csv as they are to maintain relations.

Probably is easier to get products exported as xml instead of csv and then build a tool to transform the xml to map the field names to database columns or use a simple search and replace in the xml files.
